### ISSUE TITLE
[HOLD for payment 2023-07-26] [$1000] Public room bug- Page redirects to sign in page instead of relevant page (on button click)

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#21977]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1 Create a room from user A (mobile number) in public mode
2 Go to share code and copy the link of the room
3. Open the link in Incognito mode (here incognito represents first time user)
4. Go to details and click on any button (share code, settings, Leave room) and verify if you can click on it or are redirected to sign in page
## Expected Result: 
Upon clicking any button on the detail page of the public room, the relevant page associated with that button should open. The user should be able to access the intended features or content of the room. If it's a feature, then the user should be forbidden from accessing the detail page.
## Actual Result:  
Instead of opening the relevant page associated with the button, the user is redirected to the sign-in page. This redirection prevents the user from accessing the desired content or features of the public room, causing inconvenience and hindering the intended functionality.

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.34-1
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/93bd8e20-5a76-4db4-a579-67070596e025

https://github.com/Expensify/App/assets/43996225/4b26c07a-6003-4ca5-80a0-a52aed15aa52

**Expensify/Expensify Issue URL:**
**Issue reported by:** @avi-shek-jha
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1687822639813859

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01853fdd0f3e17fe66</li>
        <li>Upwork Job ID: 1675709310305165312</li>
        <li>Last Price Increase: 2023-07-03</li>
    </ul>
</details>

**Comment by CortneyOfstad** [comment]
Heading OoO for the week (back July 10) so re-assigning this in the meantime 👍

**Comment by sophiepintoraetz** [comment]
@CortneyOfstad - please can you:
1. Keep yourself assigned
2. Leave detailed steps as to what needs to happen next 

[This SO covers it all in detail!](https://stackoverflowteams.com/c/expensify/questions/8836)

**Comment by sophiepintoraetz** [comment]
Reproduced!
![2023-07-03_15-29-57 (1)](https://github.com/Expensify/App/assets/42940078/d3744490-11d2-4ee8-9e8e-6d777f109fa1)

**Comment by Santhosh-Sellavel** [comment]
Seems @dukenv0307 Is on to the solution [here](https://github.com/Expensify/App/issues/21977#issuecomment-1615006602), will confirm tomorrow!

**Comment by Santhosh-Sellavel** [comment]
I like the idea but it doesn't work for me @dukenv0307, are you missing something in the proposal?

**Comment by Santhosh-Sellavel** [comment]
@dukenv0307's proposal looks good to me, will improve in the PR

C+ Reviewed
🎀 👀 🎀

**Comment by PauloGasparSv** [comment]
> @dukenv0307's proposal looks good to me, will improve in the PR
> 
> C+ Reviewed 🎀 👀 🎀

Agree : )

**Comment by Santhosh-Sellavel** [comment]
@PauloGasparSv This might look like a simple open-shut issue, but this could open load issues. 
We should check all the existing usage `checkIfActionIsAllowed` and restrictions based on Anonymous users and come up with what are the actions. Should decide whether we want to allow it or not, handle it in a single or multiple PR's what do you say?

Here is [another](https://github.com/Expensify/App/issues/22307) bug that shares a common root cause.

What do you say?

**Comment by Santhosh-Sellavel** [comment]
@PauloGasparSv This might look like a simple open-shut issue, but this could open load issues.
We should check all the existing usage checkIfActionIsAllowed and restrictions based on Anonymous users and come up with what are the actions. Should decide whether we want to allow it or not, handle it in a single or multiple PR's what do you say?

Here is https://github.com/Expensify/App/issues/22307 bug that shares a common root cause.

What do you say?

**Comment by PauloGasparSv** [comment]
> We should check all the existing usage checkIfActionIsAllowed and restrictions based on Anonymous users and come up with what are the actions. Should decide whether we want to allow it or not, handle it in a single or multiple PR's what do you say?

Oh, I was under the impression that adding `isAnonymousAction = false` would make sure this logic wouldn't break anything,

> We should check all the existing usage checkIfActionIsAllowed and restrictions based on Anonymous users and come up with what are the actions. Should decide whether we want to allow it or not, handle it in a single or multiple PR's what do you say?

That makes sense, I think we have ~10 calls to `checkIfActionIsAllowed` so shouldn't be too much of a problem. @dukenv0307 can you follow up to this and check how these calls are working after your fix?

I think the testing and QA steps will be too big or suck if we handle everything in a single P.R. so I'm not sure how to handle this!

@marcochavezf since you introduced this logic, just so you are aware or has anything to comment : )

**Comment by PauloGasparSv** [comment]
> @Santhosh-Sellavel and @PauloGasparSv where are we at here? Anything needed on my side? TIA!

Hey @CortneyOfstad, I think we are good! The P.R. has been constantly updated (latest update [here](https://github.com/Expensify/App/pull/22321#issuecomment-1633470891)) and receiving reviews.

Btw, [we are also including fixes](https://github.com/Expensify/App/pull/22321#issuecomment-1630903469) for https://github.com/Expensify/App/issues/22638 and https://github.com/Expensify/App/issues/22307 there.

**Comment by PauloGasparSv** [comment]
Hey just wanted to let everyone know that **I'm going OOO for 2 weeks** (starting on July 17th, back on July 31st).

I'm keeping me as the assignee here since the P.R. was merged yesterday but if an internal engineer is required please feel free to re-assign this to someone else!

**Comment by CortneyOfstad** [comment]
Thanks @PauloGasparSv — enjoy your time off!

**Comment by CortneyOfstad** [comment]
PR was deployed to staging, so just waiting to see it go into production 👍

**Comment by Santhosh-Sellavel** [comment]
This was straightforward work, the delay was to make a decision for the options. A slight scope increase, so requesting a bonus for this one! We can wait until @PauloGasparSv's back to make a decision if required, thanks!

**Comment by CortneyOfstad** [comment]
@Santhosh-Sellavel I believe that this is fine, but will wait for @PauloGasparSv to be safe 👍

**Comment by CortneyOfstad** [comment]
Things look good, so I'm going to reapply the engineering label and see if there is anything else that needs to be done, or if we can get this paid today 👍

**Comment by CortneyOfstad** [comment]
@cristipaval Paulo was originally assigned but is on vacation until July 31st. Just checking to see if you can do a quick review and see if we're set to get this paid, and if so if @Santhosh-Sellavel qualifies for the bonus. TIA!

**Comment by CortneyOfstad** [comment]
Sorry about that! Bump @cristipaval

**Comment by cristipaval** [comment]
@PauloGasparSv is back, I think he has more context on this.

**Comment by PauloGasparSv** [comment]
As @cristipaval mentioned I got back today so I can help with that :D 
I'll also check if the 2 issues that were on HOLD for this are fixed.

I just need to get some stuff done before I focus here so doing this later today or tomorrow morning!

**Comment by CortneyOfstad** [comment]
Sounds good — thanks @PauloGasparSv. Once you review, let me know so I can send out the payouts today, as I'm heading OoO tomorrow 👍

**Comment by CortneyOfstad** [comment]
Contracts have been offered to @Santhosh-Sellavel and @dukenv0307 ($1,000 with the additional 50% bonus to be applied at payment). @Santhosh-Sellavel please let me know as soon as you accept, as @dukenv0307 already has in Upwork 👍 

@avi-shek-jha, can you link your UpWork profile so I can pay you the reporting amount? Thanks!

**Comment by avi-shek-jha** [comment]
> @avi-shek-jha, can you link your UpWork profile so I can pay you the reporting amount? Thanks!

@CortneyOfstad Here is my upwork profile link: https://www.upwork.com/freelancers/~01a90e548bf808418c

Thank you

**Comment by CortneyOfstad** [comment]
@avi-shek-jha — perfect! The offer has been sent your way! Please let me know as soon as you accept 👍

**Comment by avi-shek-jha** [comment]
> @avi-shek-jha — perfect! The offer has been sent your way! Please let me know as soon as you accept 👍

@CortneyOfstad I have accepted the offer. Thank you

**Comment by PauloGasparSv** [comment]
I followed up to https://github.com/Expensify/App/issues/22307 and https://github.com/Expensify/App/issues/22638 asking for the issues to be closed, the P.R. worked and fixed them as expected : )

**Comment by Santhosh-Sellavel** [comment]
@PauloGasparSv Can you reopen I'm not paid yet.

**Comment by PauloGasparSv** [comment]
I think @CortneyOfstad may be OOO already so let me re-assign this just in case : )

**Comment by PauloGasparSv** [comment]
Hey @conorpendergrast I removed  @CortneyOfstad and assigned you here just so we can follow up to [this](https://github.com/Expensify/App/issues/21977#issuecomment-1661150059) : )

**Comment by Santhosh-Sellavel** [comment]
@conorpendergrast Can you just summarize the payment amount for me, so I can request payment on the New Dot. Everyone else is paid here. Also please close the upwork contact, thanks!

**Comment by conorpendergrast** [comment]
Payouts summary:

- [x] Issue Reporter: $250 @avi-shek-jha. Paid via Upwork
- [x] Contributor: $$1000 + $500 urgency bonus = $1500 @dukenv0307. Paid via Upwork
- [ ] Contributor+: $1000 + $500 urgency bonus = $1500 @Santhosh-Sellavel. To be paid via Manual Request

Eligible for 50% #urgency bonus? Yes

[Upwork job is here](https://www.upwork.com/ab/applicants/1675709310305165312/job-details)

**Comment by conorpendergrast** [comment]
@Santhosh-Sellavel Can you link me to the contract? I am not confident that Upwork will allow me to see/ interact with it, so that may need to wait until Cortney is back from OoO.

**Comment by conorpendergrast** [comment]
@Santhosh-Sellavel Bump on that contract link please!

**Comment by Santhosh-Sellavel** [comment]
No worries seems the offer is withdrawn. Wait until the payment request & approval thanks!

**Comment by Santhosh-Sellavel** [comment]
Requested on ND

**Comment by JmillsExpensify** [comment]
Reviewed details for @Santhosh-Sellavel. These details are accurate based on summary from Business Reviewer and are now approved for payment in NewDot.

**Comment by conorpendergrast** [comment]
@JmillsExpensify Should you tick that off in [the summary](https://github.com/Expensify/App/issues/21977#issuecomment-1661731807) too?

**Comment by conorpendergrast** [comment]
@Santhosh-Sellavel I just spotted we're missing the [C+ checklist](https://github.com/Expensify/App/issues/21977#issuecomment-1642500053) here. Once that's done and any regression tests are created, we're good to close!

**Comment by Santhosh-Sellavel** [comment]
> **BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:
> 
> - [ ] [@Santhosh-Sellavel] The PR that introduced the bug has been identified. Link to the PR:
> - [ ] [@Santhosh-Sellavel] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
> - [ ] [@Santhosh-Sellavel] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
> - [ ] [@Santhosh-Sellavel] Determine if we should create a regression test for this bug.
> - [ ] [@Santhosh-Sellavel] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
> - [ ] [@CortneyOfstad / @sophiepintoraetz] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by Santhosh-Sellavel** [comment]
Bug: Anonymous actions like copy messages, share code, copy link to the room are not allowed.

### Regression Test Steps

1. Login with any account
2. Create a public room
3. Go to share code and copy the link to the room
4. Sign out and open the public room with the link above
5. Click on the header
6. Go to share code
7. Verify that the share code page appears instead of redirecting to sign in page
8. Hover any message and click on copy to clipboard
9. Verify that the user can copy this message to clipboard and not redirect to sign page
10. Hover on an attachment and click on download
11. Verify that the user can download successfully

**Comment by conorpendergrast** [comment]
Nice! Regression test created: https://github.com/Expensify/Expensify/issues/307032

**Comment by conorpendergrast** [comment]
All done
